block/test/BlockSpec.hs

```diff
@@ -0,0 +1,67 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE ConstraintKinds #-}
+{-# LANGUAGE TypeSynonymInstances #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE FlexibleContexts     #-}
+{-# OPTIONS_GHC -Weverything -Wno-unsafe -Wno-type-defaults #-}
+
+module BlockSpec (spec, main) where
+
+import           Universum
+import           Test.Hspec
+import           Control.Monad.IO.Class  (liftIO)
+import qualified Pos.DB       as DB
+import qualified Pos.DB.Block as DB
+import           Pos.Core                (MainBlock, ProtocolMagic(..), ProtocolConstants(..), Block)
+import           Pos.Arbitrary.Block.Generate (generateMainBlock)
+import           Pos.Binary.Class (Bi, decodeFull', serialize')
+import           Pos.Core (headerHash)
+import           Pos.Block.Base
+
+pm :: ProtocolMagic
+pm = ProtocolMagic 0
+
+pc :: ProtocolConstants
+pc = ProtocolConstants
+    { pcK = 7
+    , pcVssMaxTTL = maxBound
+    , pcVssMinTTL = minBound
+    }
+
+main :: IO ()
+main = hspec spec
+
+spec :: Spec
+spec = do
+  describe "Block" $ do
+    it "put and get" $ do
```

Comment:
This is pretty similar to how I format my code outside this project, but the existing style guide does say 4 space indentation.